Refactor: Rename Orchestrator to VTOrc in the codebase

GuptaManan100 opened the pull request:
<!--
  Thank you for your contribution to the Vitess project.
  How to contribute: https://vitess.io/docs/contributing/
  Please first make sure there is an open Issue to discuss the feature/fix suggested in this PR.
  If this is a new feature, please mark the Issue as "RFC".
 -->

<!-- if this PR is Work in Progress please create it as a Draft Pull Request -->

## Description

<!-- A few sentences describing the overall goals of the pull request's commits. -->
<!-- If this is a bug fix and you think the fix should be backported, please write so. -->

This PR renames the occurrences of `orchestrator` with `vtorc` in package names, function names, comments, logs, etc.
Some usages of `orchestrator` are still left since they actually refer to the original orchestrator.

VTop changes are also required for this PR  since the web directory has also changed from `web/orchestrator` to `web/vtorc`

Configuration parameters having `Orchestrator` in their name have not been refactored yet.


## Related Issue(s)

<!-- List related issues and pull requests. If this PR fixes an issue, please add it using Fixes #????  -->

## Checklist

-   [x] "Backport me!" label has been added if this change should be backported
-   [x] Tests were added or are not required
-   [x] Documentation was added or is not required

## Deployment Notes

<!-- Notes regarding deployment of the contained body of work. These should note any db migrations, etc. -->

deepthi commented:
If someone is already running VTOrc with vitess-operator, what is the migration path for them?

GuptaManan100 commented:
There isn't. When they upgrade their operator version, they'll have to update the VTOrc image too.

GuptaManan100 commented:
One way to make it work is to copy the `web/orchestrator` directory into `web/vtorc` but not remove the first. Then we can remove it after v15 is released.

deepthi commented:
> There isn't. When they upgrade their operator version, they'll have to update the VTOrc image too.

Can we just document this in the `Deployment Notes` and release notes for both Vitess 15 and VTop 2.8 releases? That vitess and vitess-operator will need to be upgraded together?
The usual practice is to upgrade the operator and then rollout the vitess version change. Will that work?

rsajwani commented on go/vt/vtorc/db/tls.go:
```diff
@@ -124,15 +124,15 @@ func SetupMySQLTopologyTLS(uri string) (string, error) {
-func SetupMySQLOrchestratorTLS(uri string) (string, error) {
-	if !orchestratorTLSConfigured {
+func SetupMySQLVTOrcTLS(uri string) (string, error) {
+	if !vtorcTLSConfigured {
```
I am observing all the flags and function name are VTOrc then why is vtorcTLSConfigured not following this convention?

rsajwani commented on go/vt/vtorc/db/tls.go:
```diff
@@ -124,15 +124,15 @@ func SetupMySQLTopologyTLS(uri string) (string, error) {
 		tlsConfig.MinVersion = tls.VersionTLS10
 		if err != nil {
-			log.Fatalf("Can't create TLS configuration for Orchestrator connection %s: %s", uri, err)
+			log.Fatalf("Can't create TLS configuration for VTOrc connection %s: %s", uri, err)
```
VTOrc vs Vtorc

rsajwani commented on go/vt/vtorc/inst/instance_topology.go:
```diff
@@ -2070,7 +2070,7 @@ func relocateReplicasInternal(replicas []*Instance, instance, other *Instance) (
 
 // RelocateReplicas will attempt moving replicas of an instance indicated by instanceKey below another instance.
-// Orchestrator will try and figure out the best way to relocate the servers. This could span normal
+// VTOrc will try and figure out the best way to relocate the servers. This could span normal
```
VTOrc vs Vtorc

rsajwani commented on go/vt/vtorc/inst/tablet_dao.go:
```diff
@@ -45,7 +45,7 @@ var ErrTabletAliasNil = errors.New("tablet alias is nil")
 // the necessary propagation to the old primary. The propagation is best
 // effort. If it fails, the tablet's shard sync will eventually converge.
-// The proactive propagation allows a competing Orchestrator from discovering
+// The proactive propagation allows a competing VTOrc from discovering
```
VTOrc vs Vtorc

rsajwani reviewed (approved):
LGTM. Overall I am observing you are using VTOrc for functions names, variable names and flags name. But using Vtorc in comments and log statements. I did find some inconsistency with the convention so have pointed out in the review.  Feel free to correct me if my observation is not correct.

rsajwani commented:
There are some static code analysis check failures. please fix them before you  push the code

GuptaManan100 commented on go/vt/vtorc/app/cli.go:
```diff
@@ -1116,7 +1116,7 @@ func Cli(command string, strict bool, instance string, destination string, owner
 		}
 		// Recovery & analysis
-	case registerCliCommand("recover", "Recovery", `Do auto-recovery given a dead instance`), registerCliCommand("recover-lite", "Recovery", `Do auto-recovery given a dead instance. Orchestrator chooses the best course of actionwithout executing external processes`):
+	case registerCliCommand("recover", "Recovery", `Do auto-recovery given a dead instance`), registerCliCommand("recover-lite", "Recovery", `Do auto-recovery given a dead instance. VTOrc chooses the best course of actionwithout executing external processes`):
```
VTOrc in the capitalized form and vtorc in the lower cased version

GuptaManan100 commented on go/test/endtoend/vtorc/utils/utils.go:
```diff
@@ -269,7 +269,7 @@ func StopVtorcs(t *testing.T, clusterInfo *VtOrcClusterInfo) {
 }
 
-// SetupVttabletsAndVtorc is used to setup the vttablets and start the orchestrator
+// SetupVttabletsAndVtorc is used to setup the vttablets and start the vtorcs
```
Fixed function name

GuptaManan100 commented:
@rsajwani We want to use `VTOrc` for the capitalized form and `vtorc` when lowercased. I have gone ahead and changed all `Vtorc` to `VTOrc` too in the codebase. This should make everything consistent.

GuptaManan100 commented:
I have made the changes like we discussed @deepthi. I have added a symlink from `web/orchestrator` to `web/vtorc` and changed the dockerfiles to include both. So, the changes are now non-breaking. I have tested the new built image with vitess-operator and it works without any changes. I have also checked the compose example and local examples, to verify that they continue to work.

deepthi reviewed (approved):
LGTM.
> I have tested the new built image with vitess-operator and it works without any changes. I have also checked the compose example and local examples, to verify that they continue to work.

Thank you for the diligence in testing

rsajwani commented:
> @rsajwani We want to use `VTOrc` for the capitalized form and `vtorc` when lowercased. I have gone ahead and changed all `Vtorc` to `VTOrc` too in the codebase. This should make everything consistent.

Ah make sense. Thanks

matthiasr commented:
Or will the operator understand the difference and handle it automatically?
